src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/jpa/repository/TierCalculationRepository.kt

```diff
@@ -13,6 +14,9 @@ interface TierCalculationRepository : CrudRepository<TierCalculationEntity, Long
 
   fun findByCrnAndUuid(crn: String, calculationId: UUID): TierCalculationEntity?
 
-  @Query("SELECT DISTINCT crn FROM TierCalculationEntity ORDER BY crn DESC")
-  fun findDistinctCrn(): List<String>
+  @Query(nativeQuery = true, value = "SELECT DISTINCT crn FROM tier_calculation ORDER BY crn DESC OFFSET ?1 LIMIT ?2")
```

Comment:
you could have used `Pageable` and return a `Page` here both things Spring Data has out of the box https://stackoverflow.com/questions/22345081/spring-data-jpa-query-and-pageable so that you can get the total pages to query

Reply:
yes, I think having a limit/offset will give us more control on specific number of rows we want.

Reply:
you can supply a size with pageable and then page number does the offset, the comment was more of an FYI for the future when doing pagination